app-foundation/4-projects/non-production/boa_secret.tf

```diff
@@ -0,0 +1,135 @@
+    "${module.boa_gke_project.project_id}=>roles/iam.serviceAccountAdmin",
+    "${module.boa_gke_project.project_id}=>roles/iam.serviceAccountUser",
+    "${module.boa_gke_project.project_id}=>roles/resourcemanager.projectIamAdmin",
+    "${module.boa_gke_project.project_id}=>roles/logging.configWriter",
+    "${module.boa_gke_project.project_id}=>roles/storage.objectViewer",
+    "${module.boa_gke_project.project_id}=>roles/iap.admin",
+    "${module.boa_gke_project.project_id}=>roles/iam.roleAdmin",
+    "${module.boa_ops_project.project_id}=>roles/logging.configWriter",
+    "${module.boa_ops_project.project_id}=>roles/serviceusage.serviceUsageAdmin",
+    "${module.boa_ops_project.project_id}=>roles/resourcemanager.projectIamAdmin",
+    "${module.boa_ops_project.project_id}=>roles/storage.admin",
+    "${module.boa_secret_project.project_id}=>roles/cloudkms.admin",
+    "${module.boa_secret_project.project_id}=>roles/logging.configWriter",
+    "${module.boa_secret_project.project_id}=>roles/iam.serviceAccountCreator",
+    "${module.boa_secret_project.project_id}=>roles/secretmanager.admin",
+    "${module.boa_sql_project.project_id}=>roles/cloudsql.admin",
+    "${module.boa_sql_project.project_id}=>roles/compute.networkAdmin",
+    "${module.boa_sql_project.project_id}=>roles/logging.configWriter",
+    "${var.shared_vpc_host_project_id}=>roles/compute.networkAdmin",
+    "${var.shared_vpc_host_project_id}=>roles/compute.securityAdmin"
+  ]
+}
+
+resource "google_service_account_iam_member" "cloudbuild_terraform_sa_impersonate_permissions" {
+  service_account_id = module.terraform_deployment_sa.service_account.name
+  role               = "roles/iam.serviceAccountTokenCreator"
+  member             = "serviceAccount:${var.app_infra_pipeline_cloudbuild_sa}"
+}
+
+data "google_projects" "shared_vpc_project" {
+  filter = "name:${var.project_prefix}-${var.environment_code}-shared-base"
+}
+
+data "google_compute_network" "vpc" {
+  project = var.shared_vpc_host_project_id
+  name    = var.shared_vpc_network_name
+}
+
+data "google_compute_subnetwork" "subnet" {
+  for_each  = toset(data.google_compute_network.vpc.subnetworks_self_links)
+  self_link = each.value
+}
+
+resource "google_compute_subnetwork_iam_member" "terraform_subnet_member" {
+  for_each   = data.google_compute_subnetwork.subnet
+  project    = each.value.project
+  region     = each.value.region
+  subnetwork = each.value.name
+  role       = "roles/compute.networkUser"
+  member     = module.terraform_deployment_sa.iam_email
+}
+
+resource "google_compute_subnetwork_iam_member" "gke_subnet_member" {
+  for_each   = data.google_compute_subnetwork.subnet
+  project    = each.value.project
+  region     = each.value.region
+  subnetwork = each.value.name
+  role       = "roles/compute.networkUser"
+  member     = "serviceAccount:service-${module.boa_gke_project.project_number}@container-engine-robot.iam.gserviceaccount.com"
```

Comment:
Since we create the project as [service projects](https://github.com/terraform-google-modules/terraform-example-foundation/blob/c2b8200e49a718febc071d17c40ea6dfe1e49578/4-projects/modules/single_project/main.tf#L33), these should automatically be granted by [PF](https://github.com/terraform-google-modules/terraform-google-project-factory/blob/1bde0af3d3087dbd28c5b8bf296fccac094499e4/main.tf#L74). Is that not the case?

Reply:
yes seems it should be, but since we deployed a new vpc in the deployment prjs after PF had already run, the iam_binding did not exist and I explicitly added it as this is critical to get GKE deployed.
I can take this out with the assumption 3-networks will do things correctly though, leaving it in does no harm as well as I use iam_member which is Non-authoritative

Reply:
lets open an issue to track and keep this in for now